rust/src/runner/context.rs

```diff
@@ -228,3 +228,20 @@ impl TestContext {
         thread::sleep(self.naptime)
     }
 }
+
+// We implement Drop to circumvent the errors from unlinking a directory for which
+// search or write permission is denied.
+impl Drop for TestContext {
+    fn drop(&mut self) {
+        let iter = walkdir::WalkDir::new(self.base_path()).into_iter();
+        for entry in iter {
+            let entry = match entry {
+                Ok(e) => e,
+                Err(_) => continue,
+            };
+
+            let write_perm = Mode::from_bits_truncate(0o700);
+            let _ = lchmod(entry.path(), write_perm);
```

Comment:
This is probably slower than first `stat`ing the file to check it for writability.  But probably fast enough.  A bigger concern is for the ENOSPC tests.  With ZFS, it's possible that the chmod operation could fail.

Reply:
It seems that these tests don't change the permissions, however. I'm unsure of how we should handle errors in the destructor. Maybe print a warning for the user when it fails?

Reply:
There isn't a whole lot that you can do there.  I think it's ok as is, but you could also print a warning.  The important thing is that you don't panic if the thread is already handling a panic.

Reply:
Alright, I will keep it as it then.